toggl/api/models.py

```diff
@@ -623,15 +625,21 @@ def all_from_reports(self, start=None, stop=None, workspace=None, config=None):
         :return: Generator that yields TimeEntry
         """
         from .. import toggl
+
         config = config or utils.Config.factory()
         page = 1
 
-        try:
+        if workspace is None:
```

Comment:
Why are you making this refactor? I believe that you are not changing anything in the resulting algorithm no? 

And Python is very much duck-typing language so I prefer EAFP (Easier to Ask Forgiveness than Permission) rather than the LBYL (Look Before You Leap).

Reply:
Just following what I've gathered is standard practice and easiest to read. The original code was confusing and led to it being a harder to follow what was going wrong here in the first place.

Checking types is the common thing to do when your argument is a Union of several types, and has the benefit that it can be picked up by typecheckers like mypy to correctly infer the type.

This has nothing to do with EAFP as I interpret it, and more about "explicit is better than implicit". It's a lot harder to read code which raises exceptions caused by invalid types than code which uses simple type checks.

Reply:
While I get your point, I don't agree as for me this is exactly the case of LBYL. Yet, I won't be blocking this for that.